dynamic-demo-plugin/src/components/k8sConsumer/K8sAPIConsumer.tsx

```diff
@@ -10,11 +10,13 @@ import {
   PageSection,
   Title,
 } from "@patternfly/react-core";
-import { k8sCreate, k8sDelete, k8sGet, k8sList, k8sPatch, K8sResourceCommon, k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
-import { DeploymentModel, mockDeploymetData } from './k8s-data';
+import { getGroupVersionKindForResource, getReference, k8sCreate, k8sDelete, k8sGet, k8sList, k8sPatch, K8sResourceCommon, k8sUpdate, useK8sModel } from '@openshift-console/dynamic-plugin-sdk';
+import { mockDeploymetData } from './k8s-data';
 
 const K8sAPIConsumer: React.FC = () => {
   const { t } = useTranslation("plugin__console-demo-plugin");
+  const {group, version, kind} = getGroupVersionKindForResource(mockDeploymetData);
+  const [k8sModel] = useK8sModel(getReference(group, version, kind));
```

Comment:
if you change the type as suggested here https://github.com/openshift/console/pull/10303#discussion_r740175385

we would be able to do 
```suggestion

  const [k8sModel] = useK8sModel(getReference(getGroupVersionKindForResource(mockDeploymetData)));
```